pkg/resource/deploy/step.go

```diff
@@ -667,6 +671,15 @@ func (s *RefreshStep) Apply(preview bool) (resource.Status, StepCompleteFunc, er
 		}
 		if initErr, isInitErr := err.(*plugin.InitError); isInitErr {
 			initErrors = initErr.Reasons
+
+			// Partial failure SHOULD NOT cause refresh to fail. Instead:
+			//
+			// 1. Warn instead that during refresh we noticed the resource has become unhealthy.
+			// 2. Make sure the initialization errors are persisted in the state, so that the next
+			//    `pulumi up` will surface them to the user.
+			err = nil
+			msg := fmt.Sprintf("Refreshed resource is in an unhealthy state:\n* %s", strings.Join(initErrors, "\n* "))
```

Comment:
What does "unhealthy state" mean?  Is that a term we use anywhere else to refer to this?  

Is there an action we expect users to be able to take here?  If not, why a warning?  If os, can we be more clear on what the problem is and what the user should consider doing about it?

What is the semantic meaning of an `InitError` coming from `Read`?  Read itself is not responsible for "initializing" anything, so it's in a sense surprising that callers of read should be expected to handle this at all.  Why is this not something that the provider should have just not returned as an error in the first place?